pkg/controller/upgrade_funcs.go

```diff
@@ -540,171 +305,351 @@ func CloudletAccessVars(ctx context.Context, objStore objstore.KVStore, allApis
 	return nil
 }
 
-func AddStaticFqdn(ctx context.Context, objStore objstore.KVStore, allApis *AllApis, sup *UpgradeSupport) error {
-	// 1. Update cloudlets - set StaticRootLbFqdn
-	cloudletKeys, err := getDbObjectKeys(objStore, "Cloudlet")
+// InstanceKeysRegionScopedName deals with moving the CloudletKey out
+// of the AppInst/ClusterInst Key and onto the object body, changing
+// the unique key. Also, this requires the instance name to be unique
+// within the entire region instead of within a cloudlet scope.
+func InstanceKeysRegionScopedName(ctx context.Context, objStore objstore.KVStore, allApis *AllApis, sup *UpgradeSupport) error {
+	// Unfortunately the last upgrade func had a bug which failed to
+	// asign object IDs to AppInsts and ClusterInsts. We run the fixed
+	// version here again. Note that all upgrade functions must be idempotent.
+	err := UpgradeCrmOnEdge(ctx, objStore, allApis, sup)
```

Comment:
Did you actually upgrade any setups already? If not, a fixed `UpgradeCrmOnEdge` would take care of things, but no harm in calling it again :)

Reply:
I guess no setups are upgraded, although someone could technically pull up to a certain commit and upgrade on their own. I think it's safer to call it again.